py-polars/polars/dataframe/frame.py

```diff
@@ -7262,7 +7262,11 @@ def quantile(
         return self._from_pydf(self._df.quantile(quantile, interpolation))
 
     def to_dummies(
-        self, columns: str | Sequence[str] | None = None, *, separator: str = "_"
+        self,
+        columns: str | Sequence[str] | None = None,
+        *,
+        separator: str = "_",
+        include_null: bool = False,
```

Comment:
This should be `True` by default.

Reply:
Note that this will still be a breaking change since the `null` column will be included even if there is no `null` value in the data.

Reply:
It shouldn't. It should have no special treatment.

Reply:
I'm not super convinced that the current behavior should be the default: often, `to_dummies` is used for training/prediction of a downstream model. In this instance, having a predictable number of columns is rather important.

If I want to have a `null` column regardless of whether I have `null` values in my data, I'd need to manually check if there are any `null` values in my data and - if not - add an additional empty column. It would be much nicer if this was handled directly in `to_dummies`.

In theory, we could also define `include_null: Literal["auto"] | bool = "auto"` which retains the current behavior but allows for predictable outcomes.

Reply:
Alternatively, we can also close this one and continue with #7947 which allows to manually specify the expected columns. I think it already works to pass `None` to the list of values there.

Reply:
>If I want to have a null column regardless of whether I have null values in my data, I'd need to manually check if there are any null values in my data and - if not - add an additional empty column. It would be much nicer if this was handled directly in to_dummies.

Then you overwrite the null column if it doesn't exist. The null should not have a special treatment.

Reply:
@borchero If you're working with linear models, and it's the case that different datasets can have different levels of categorical variables, you should *not* be adding empty columns to make them have the same columns.
What you should be doing is as follows:
- Store `model.coef` as a DataFrame with feature names as columns.
- Code `model.predict(X)` as `X[common_features] @ model.coef[common_features]` (where the exact details of the matrix multiplication are left to you).

Reply:
@s-banach I think this is mostly unrelated to this PR. The rationale behind consistency is that the set of categorical variables you observe at inference might not be the entire set of categorical variables known at training time.

Consider the extreme case of performing inference with a single data sample: you will only observe one category but you need to add zero columns for all categories that you've seen during training (or rather, the ones you decided to include in your model).

In any case, that's not the issue that this PR attempts to solve.